compiler/builtins/string.ts

```diff
@@ -1232,12 +1232,12 @@ export const __String_prototype_split = (_this: string, separator: any, limit: a
   if (Porffor.wasm`local.get ${limit+1}` == Porffor.TYPES.undefined) limit = Number.MAX_SAFE_INTEGER;
   limit |= 0;
   if (limit < 0) limit = Number.MAX_SAFE_INTEGER;
-
   let tmp: string = Porffor.allocate(), tmpLen: i32 = 0;
   const thisLen: i32 = _this.length * 2, sepLen: i32 = separator.length;
-  if (sepLen == 1) {
+  if (sepLen == 1 || separator === undefined) {
     // fast path: single char separator
-    const sepChar: i32 = separator.charCodeAt(0);
+    const sepChar: i32 = separator ? separator.charCodeAt(0) : 0 ;
```

Comment:
This would separate null, a separate path for no separator given is probably needed

Reply:
Fixed

Reply:
It looks the same in the latest diff? (Ignore if you haven't pushed it yet)

Reply:
I was supposed to handle for null separator ? no ?

Reply:
I mean it would separate the null character (`'\00'`), not when given `null`, sorry for the misunderstanding.

Reply:
You can just do a separate path for if the separate is undefined with something like
```ts
out.length = 1;
out[0] = _this;
return out;
```

Reply:
I think there is a bug with the array implementation, because i get `undefined` in the first index

I had tried this approach before as well, it was returning `[undefined]`, and then i had to reuse the same code for single char separator. 

```js
  if (separator === undefined || separator === null) {
    out.length = 1;
    out[0] = _this;
    Porffor.print(out[0]);
    return out;
  }
  ```
  
  when i print out[0], then it is showing the string, but when i print the out array, it returns `[undefined]`
  
  Am i missing anything ?

Reply:
Yeah that is not right, not your fault, I'll look into it